gitlabform/gitlab/schedules.py

```diff
@@ -0,0 +1,117 @@
+from gitlabform.gitlab.core import GitLabCore
+
+
+class GitLabPipelineSchedules(GitLabCore):
+    def get_all_pipeline_schedules(self, project_and_group_name):
+        return self._make_requests_to_api(
+            "projects/%s/pipeline_schedules", project_and_group_name
+        )
+
+    def get_pipeline_schedule(self, project_and_group_name, pipeline_schedule_id):
+        return self._make_requests_to_api(
+            "projects/%s/pipeline_schedules/%s",
+            (project_and_group_name, pipeline_schedule_id),
+        )
+
+    def create_pipeline_schedule(
+        self, project_and_group_name, description, ref, cron, optional_data=None
+    ):
+        if optional_data is None:
```

Comment:
Nit: isn't it enough to set the `optional_data` default value to `{}`? I think that then we can get rid of this `if`.

The same would apply to similar cases below.

Reply:
This is here because IDE suggested to me that having mutable arguments is actually quite error-prone pattern.

Leaving explanation example for your reference: https://florimond.dev/blog/articles/2018/08/python-mutable-defaults-are-the-source-of-all-evil/